### PR TITLE
111 get chatroom

### PR DESCRIPTION
**Pull request by DLiSWE**
get all chatrooms by userid

### CODE REVIEW

File: src/controllers/chatroom.controller.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { RequestWithUser } from '@/interfaces';
+import { NextFunction, Response } from 'express';
+import { ChatroomService } from '@/services';
+import { BaseChatroomDto } from '@/dtos';
+
+class ChatroomController {
+  private chatroomService = new ChatroomService();
+
+  public getAllChatrooms = async (req: RequestWithUser, res: Response, next: NextFunction) => {
```
Comment (emir4ik-code) [inline src/controllers/chatroom.controller.ts:9]: Nit: Maybe we can call it "getAllUserChatrooms"

**Review by emir4ik-code — APPROVED** [review]
consider variable naming suggestions, otherwise looks good.

File: src/services/chatroom.service.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { BaseChatroomDto } from '@/dtos';
+import { ChatroomSchema } from '@/entities/nosql';
+import { HttpException } from '@/exceptions';
+import { model } from 'mongoose';
+
+class ChatroomService {
+  public async getAllChatrooms(userId: number): Promise<BaseChatroomDto[]> {
+    const chatroom = model('Chatroom', ChatroomSchema);
+    const getAllChat: BaseChatroomDto[] = await chatroom.find({ users: { $in: userId } });
+    if (getAllChat.length === 0) throw new HttpException(404, 'No Chats found');
```
Comment (ZohaibManzoor00) [inline src/services/chatroom.service.ts:10]: nit:  'No chats found'. Lowercase the c in chats

**Review by ZohaibManzoor00 — APPROVED** [review]
Looks good

File: src/services/chatroom.service.ts [hunk 3]
```diff
@@ -16,6 +16,12 @@ class ChatroomService {
       createdDate: Date.now(),
     }).save();
   }
+  public async getAllChatrooms(userId: number): Promise<BaseChatroomDto[]> {
+    const chatroom = model('Chatroom', ChatroomSchema);
+    const getAllChat: BaseChatroomDto[] = await chatroom.find({ users: { $in: userId } });
```
Comment (Matt-Tan15) [inline src/services/chatroom.service.ts:21]: what's $in?

**Review by Matt-Tan15 — APPROVED** [review]

**Review by emir4ik-code — APPROVED** [review]

**Pull request merged**